### ISSUE TITLE
Vim9: inconsistent rules for space after command followed by bar

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#8592]
**Describe the bug**

In Vim9 script, the rules which enforce whitespace after a command name followed by a bar are inconsistent.

**To Reproduce**

Run this shell command:

    vim -Nu NONE -S <(cat <<'EOF'
        vim9script
        echo 'a'|echo 'b'
        echo 'no error'
    EOF
    )

No error is raised.

Now, run this other shell command:

    vim -Nu NONE -S <(cat <<'EOF'
        vim9script
        try|eval 0|catch|endtry
    EOF
    )

`E1144` is raised:

    E1144: Command "try" is not followed by white space: try|eval 0|catch|endtry

**Expected behavior**

Not sure.  Either the first command should give an error, like the second one.  Or the second command should not give an error, like the first one.

**Environment**

 - Vim version: 8.2 Included patches: 1-3187
 - OS: Ubuntu 20.04.2 LTS
 - Terminal: XTerm(353)

**Additional context**

I had a look at the tests about `E1144`; they all seem related to an inline comment:
```vim
vim9script
echo# something
```
    E1144: Command "echo" is not followed by white space: echo# something
```vim
vim9script
try# comment
    echo 'yes'
catch
endtry
```
    E1144: Command "try" is not followed by white space: try# comment

But none of them seem to test a bar; so, I don't know whether it's meant to raise an error for a bar.

**Comment by brammool** [comment]
> **Describe the bug**
> 
> In Vim9 script, the rules which enforce whitespace after a command name followed by a bar are inconsistent.
> 
> **To Reproduce**
> 
> Run this shell command:
> 
>     vim -Nu NONE -S <(cat <<'EOF'
>         vim9script
>         echo 'a'|echo 'b'
>         echo 'no error'
>     EOF
>     )
> 
> No error is raised.
> 
> Now, run this other shell command:
> 
>     vim -Nu NONE -S <(cat <<'EOF'
>         vim9script
>         try|eval 0|catch|endtry
>     EOF
>     )
> 
> `E1144` is raised:
> 
>     E1144: Command "try" is not followed by white space: try|eval 0|catch|endtry
> 
> **Expected behavior**
> 
> Not sure.  Either the first command should give an error, like the second one.  Or the second command should not give an error, like the first one.
> 
> **Environment**
> 
>  - Vim version: 8.2 Included patches: 1-3187
>  - OS: Ubuntu 20.04.2 LTS
>  - Terminal: XTerm(353)
> 
> **Additional context**
> 
> I had a look at the tests about `E1144`; they all seem related to an inline comment:
> ```vim
> vim9script
> echo# something
> ```
>     E1144: Command "echo" is not followed by white space: echo# something
> ```vim
> vim9script
> try# comment
>     echo 'yes'
> catch
> endtry
> ```
>     E1144: Command "try" is not followed by white space: try# comment
> 
> But none of them seem to test a bar; so, I don't know whether it's
> meant to raise an error for a bar.

Perhaps this check is a bit too strict.  This works:

	try |echo 0|catch /pat/|endtry

Removing /pat/ still works:

	try |echo 0|catch |endtry

But removing the space after "catch" does not:

	try |echo 0|catch|endtry

Without the space the command is still quite readable.  And we don't
insist on the white before bar in other places, so for consistency I
would drop that requirement for :try and :catch.

-- 
hundred-and-one symptoms of being an internet addict:
190. You quickly hand over your wallet, leather jacket, and car keys
     during a mugging, then proceed to beat the crap out of your
     assailant when he asks for your laptop.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by lacygoill** [comment]
Fixed by 8.2.3189.
